static/js/message_fetch.js

```diff
@@ -54,11 +54,7 @@ function process_result(data, opts) {
         message_util.add_old_messages(messages, opts.msg_list);
     }
 
-    if (
-        opts.msg_list === message_lists.current &&
-        opts.msg_list.narrowed &&
-        opts.msg_list.empty()
-    ) {
+    if (message_lists.current.empty()) {
```

Comment:
I think you still need to check that `opts.msg_list === message_lists.current` -- if we're processing messages for a view that's not currently visible, we don't want to show this banner in the currently visible view.

Reply:
Will try this out and see if it fixes the bugs I was having!